modules/ogg/ogg_packet_sequence.cpp

```diff
@@ -159,7 +159,9 @@ bool OggPacketSequencePlayback::next_ogg_packet(ogg_packet **p_packet) const {
 
 	*p_packet = packet;
 
-	packet_cursor++;
+	if (!packet->e_o_s) { // Added this so it doesn't try to go to the next packet if it's the last packet of the file.
```

Comment:
This change makes it such that this function will NEVER return false; at EOS, the packet_cursor always stays pointed to the last packet, so this will keep returning the last packet forever.

Reply:
That may be a bit unintuitive yes. Normally, callers check the `e_o_s` flag and are supposed to stop reading at the end of the stream. If they don't, then they would have previously gotten false and logged an error, whereas now, they get true and the last packet again. This doesn't make a difference as long as callers correctly check the `e_o_s` flag, but you might argue that the previous way of erroring out when trying to read past the end of the stream was the safer route.

I think this change wasn't directly related to the fix but rather one of a few modifications made in an attempt to narrow down the original problem.

Reply:
Returning a boolean value instead of an error makes it look like one of those classic reader functions that you put this in the condition of a while loop and keep going until it returns false; the fact that it doesn't behave that way is very confusing, and I ended up banging my head against the wall for a couple of hours trying to figure out why this kept not returning false.

In either case, I've tested removing this check with the example projects for the issues that this PR is purported to have fixed, and they work fine.

Reply:
Can you provide an ogg/vorbis file that causes this issue? Here or in the comments of #85996

Reply:
It's not really a problem with any particular ogg file, it's an issue with this being part of the public API and it having confusing behavior. It's also semantically incorrect, since there's no case where this function will actually return false.

When something other than godot uses this function (e.g a custom module) and expects the `next_ogg_packet()` to return false when there's no more packets to be read (e.g. `while (playback->next_ogg_packet(&pkt))`, this can cause an infinite loop.

Reply:
Okay, you mentioned an infinite loop so I assumed it was a reproducible case rather than an API issue. Sorry for misunderstanding.